matcher/core.py

```diff
@@ -118,45 +118,61 @@ def run(self):
 
         self.logger.debug('Start encoding')
 
-        encoder = Encoder(
-            reviewers=self.datasource.reviewers,
-            papers=self.datasource.papers,
-            constraints=self.datasource.constraints,
-            scores_by_type=self.datasource.scores_by_type,
-            weight_by_type=self.datasource.weight_by_type,
-            normalization_types=self.datasource.normalization_types,
-            logger=self.logger
-        )
-
-        self.logger.debug('Preparing solver')
-
-        # solver
-        solver = self.solver_class(
-            self.datasource.minimums,
-            self.datasource.maximums,
-            self.datasource.demands,
-            encoder,
-            logger=self.logger
-        )
-
-        solution = None
-        start_time = time.time()
+        try:
```

Comment:
why do you need different try/catch and not a single catch at the end of the function?

Reply:
Fixed